cache/cacheable_filter.go

```diff
@@ -0,0 +1,18 @@
+package cache
+
+import "github.com/btcsuite/btcutil/gcs"
+
+// CacheableFilter is a wrapper around Filter type which provides a Size method
+// used by the cache to target certain memory usage.
+type CacheableFilter struct {
+	*gcs.Filter
+}
+
+// Size returns size of this filter in bytes.
+func (c *CacheableFilter) Size() (uint64, error) {
+	f, err := c.Filter.NBytes()
```

Comment:
I'm not sure if we should use `Bytes`, `PBytes `, `NBytes` or `NPBytes` here, or something else entirely. One thing to think about is that these all serializes the `Filter`, which might be expensive, but I think it's better to optimize this at a later point if it turns out to be a bottleneck.

I'm leaning towards just using `Bytes` for simplicity.

Reply:
FWIW, it's serialized on the wire as `NBytes`.

Reply:
I'm fine with sticking to `NBytes` as it it just an estimate on mem usage anyway.